clippy_dev/src/main.rs

```diff
@@ -0,0 +1,45 @@
+extern crate clap;
+extern crate clippy_dev;
+extern crate regex;
+
+use clap::{App, Arg, SubCommand};
+use clippy_dev::*;
+
+fn main() {
+    let matches = App::new("Clippy developer tooling")
+        .subcommand(
+            SubCommand::with_name("update_lints")
+                .about("Update the lint list")
+                .arg(
+                    Arg::with_name("print-only")
+                        .long("print-only")
+                        .short("p")
+                        .help("Print a table of lints to STDOUT. Does not modify any files."),
+                )
+        )
+        .get_matches();
+
+    if let Some(matches) = matches.subcommand_matches("update_lints") {
+        if matches.is_present("print-only") {
+            print_lints();
+        }
+    }
```

Comment:
we should probably error if bad arguments are given, otherwise a typo can result in very confusing behaviour

Reply:
clap is already handling this for us before that part of the code is reached. Calling `get_matches()` will fail the program if incorrect arguments are supplied:
![selection_048](https://user-images.githubusercontent.com/2042399/45082721-d7dd1f80-b0f1-11e8-98b5-fb99289b63a1.png)
![selection_049](https://user-images.githubusercontent.com/2042399/45082892-44f0b500-b0f2-11e8-86c9-17c2a0d1821b.png)